raiden/transfer/channel.py

```diff
@@ -1354,6 +1518,49 @@ def events_for_close(
     return events
 
 
+def events_for_expired_lock(
+        channel_state: NettingChannelState,
+        secrethash: typing.SecretHash,
+        locked_lock: LockedTransferUnsignedState,
+        pseudo_random_generator: random.Random,
+):
+    nonce = get_next_nonce(channel_state.our_state)
+    locked_amount = get_amount_locked(channel_state.our_state)
+    our_balance_proof = channel_state.our_state.balance_proof
+    transferred_amount = our_balance_proof.transferred_amount
+    sender_state = channel_state.partner_state
+    merkletree = compute_merkletree_without(sender_state.merkletree, locked_lock.lockhash)
+    locksroot = merkleroot(merkletree)
+
+    balance_proof = BalanceProofUnsignedState(
+        nonce=nonce,
+        transferred_amount=transferred_amount,
+        locked_amount=locked_amount,
+        locksroot=locksroot,
+        token_network_identifier=channel_state.token_network_identifier,
+        channel_identifier=channel_state.identifier,
+        chain_id=channel_state.chain_id,
+    )
+
+    delete_secrethash_endstate(channel_state.our_state, secrethash)
```

Comment:
Moving this here is something you requested. But i don't see how the new state is going to become the latest if we don't return a `TransitionResult(channel_state... etc)`. Because when this function is called from `handle_block` in `mediator.py` and `target.py` the process is separate from handling a `state_change` for channels.